src/hash_picker.cpp

```diff
@@ -206,12 +206,14 @@ bool validate_hash_request(hash_request const& hr, file_storage const& fs)
 			int const file_first_piece = int(m_files.file_offset(fidx) / m_files.piece_length());
 			int const num_layers = file_num_layers(fidx);
 			int const piece_tree_root_layer = std::max(0, num_layers - m_piece_tree_root_layer);
-			auto& f = m_piece_hash_requested[fidx];
 
-			for (int i = 0; i < int(f.size()); ++i)
+			int i = -1;
```

Comment:
this part is unrelated to variable shadowing. It's meant to simplify this loop with a range-based one, and not use the index quite as much.